CONTRIBUTING.md

```diff
@@ -0,0 +1,130 @@
+# Contribute to StreamFlow
+
+As a first step, get StreamFlow from [GitHub](https://github.com/alpha-unito/streamflow) 
+```bash
+git clone git@github.com:alpha-unito/streamflow.git
+```
+
+Then, you can install all the required packages using the `pip install` command.
+
+```bash
+cd streamflow
+pip install -r requirements.txt
+```
+
+Finally, you can install StreamFlow with the following command 
+```
+pip install .
+```
+
+## Continuous Integration
+
+StreamFlow relies on [GitHub Actions](https://github.com/features/actions) for Continuous Integration (CI) and Continuous Distribution (CD).
+The maintainers take care of the CD pipeline. In order to publish a new version of the software on PyPI and Docker Hub distributions, the maintainer only has to augment the version number in the `version.py` file.
+
+Instead, everyone in the community can contribute to the StreamFlow codebase by opening a Pull Request (PR). Running the entire suite of StreamFlow tests is part of the CI pipeline.
+However, it is suggested that some tests be done locally before opening the PR, below it is explained how to do it.
+
+
+### CWL conformance
+StreamFlow complies with all stable versions of the [Common Workflow Language](https://www.commonwl.org/) (CWL) open standard (1.0, 1.1, 1.2). Plus, it implements the `cwltool:Loop` extension (see [here](https://cwltool.readthedocs.io/en/latest/loop.html)).
+You can check if your PR does not compromise CWL compliance by running the CWL conformance tests suite, using the following command
+```bash
+./cwl-conformance-test.sh
+```
+
+### StreamFlow tests
+It is necessary that the contribution not introduce some errors in the current software features.
+Some regression tests are supplied using [pytest](https://docs.pytest.org/en/7.3.x/getting-started.html). Before it is necessary to install the required packages for the testing
+```bash
+pip install -r test-requirements.txt
+```
+
+You can execute all the available tests with the command
+```bash 
+make test
+```
+Otherwise, specific tests with the command
```

Comment:
```suggestion
Otherwise, you can execute only a specific test file. For example. to verify compliance with the `cwltool:Loop` CWL extension you can use the following command
```